### PR TITLE
partner_contact_sale_info_propagation: New module

### PR DESCRIPTION
**Pull request by ernestotejeda**
Cc @Tecnativa

### CODE REVIEW

File: partner_contact_sale_info_propagation/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                childs = record.mapped('child_ids').filtered(
+                    lambda r: not r.team_id or r.team_id == record.team_id)
+                if childs:
+                    childs.write({'team_id': vals['team_id']})
+        return super(ResPartner, self).write(vals)
+
+    @api.model
+    def create(self, vals):
+        if vals.get('parent_id'):
+            if not vals.get('user_id'):
```
Comment (Nikul-Chaudhary) [inline partner_contact_sale_info_propagation/models/res_partner.py]: @ernestotejeda IMO is  better
```
 if not vals.get('user_id') and not vals.get('team_id'):
    vals.update({
        'user_id': self.browse(vals['parent_id']).user_id.id,
        'team_id': self.browse(vals['parent_id']).team_id.id,
    })
```

File: partner_contact_sale_info_propagation/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        return super(ResPartner, self).create(vals)
+
+    @api.onchange('parent_id')
+    def onchange_parent_id(self):
+        """Change Salesperson or Sales Channel if the parent company changes
+        and there's no Salesperson or Sales Channel defined yet"""
+        res = super(ResPartner, self).onchange_parent_id()
+        if self.parent_id and self.parent_id != self:
+            parent = self.parent_id
+            if not self.user_id:
```
Comment (Nikul-Chaudhary) [inline partner_contact_sale_info_propagation/models/res_partner.py:44]: same here

**Review by Nikul-Chaudhary — CHANGES_REQUESTED** [review]
Improve Code

File: partner_contact_sale_info_propagation/models/res_partner.py (hunk 1 above)
Comment (pedrobaeza) [reply partner_contact_sale_info_propagation/models/res_partner.py]: This should be:
```python
if 'user_id' not in vals and 'team_id' not in vals:
```
or empty fields won't be propagated.

Comment (pedrobaeza) [reply partner_contact_sale_info_propagation/models/res_partner.py]: But any way, it's correct to split the `if`, as there are 2 conditions inside.

Comment (Nikul-Chaudhary) [reply partner_contact_sale_info_propagation/models/res_partner.py]: Ok, No Problem @pedrobaeza ,Sound Good

**Review by cubells — APPROVED** [review]
:+1: 

LGTM

Code review and test

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**